substreams/ethereum-bancor/src/erc20_transfer.rs

```diff
@@ -0,0 +1,32 @@
+use crate::abi::erc20::events::Transfer;
```

Comment:
Yes, it has a problem with custom tokens like DAI, because it does not use the standard ERC20 method;
The signature is something like: Transfer(src, dst, amount) and not Transfer(from, to, value)

Therefore I used a custom wrapper to check instead the event as an Array, you can have a look in the function under:
// Check erc20 transfers

Reply:
> Yes, it has a problem with custom tokens like DAI, because it does not use the standard ERC20 method;
> The signature is something like: Transfer(src, dst, amount) and not Transfer(from, to, value)

I don't think this statement is correct, what matters in the signature is the types. So the two example above lead to `Transfer(address,address,u256)` and therefore have the same `topic0`


> So I don't know how to work this around with extract_balance_deltas_from_tx
I didn't get what's the issue there. Can't you do the same thing in the closure to check that the token is indeed one of those that we want? I've done something similar for Curve 

https://github.com/propeller-heads/tycho-protocol-sdk/blob/9561cc9b8abd7b3d75045163976ca516626b6714/substreams/ethereum-curve/src/modules.rs#L153-L164

Reply:
Solved